var/spack/repos/builtin/packages/grpc/package.py

```diff
@@ -47,6 +47,8 @@ class Grpc(CMakePackage):
     depends_on('c-ares')
     depends_on('abseil-cpp', when='@1.27:')
     depends_on('re2', when='@1.33.1:')
+    with when('+shared'):
+        depends_on('re2+shared')
```

Comment:
Apart from not seeing how this solves the problem, this also should be simplified to `depends_on('...', when='...')`, and it's making this package depend on re2 for *any* version when +shared, whereas above it says only to pull in re2 for 1.33.1 and above.

Reply:
Ah okay, in that case you want to make sure that re2 is built with -fPIC, so you could add a `pic` variant, default it to true, and have it add `-CMAKE_POSITION_INDEPENDENT_CODE=ON`. Also add a conflict for ~pic and +shared on re2, since cmake always enabled pic for shared libs.